note `RunAsAny` strategy in SCC

DylanLovesCoffee opened the pull request:
### What does this PR do?
Suggest that in versions of Openshift where the Docker socket is owned by the root group, we can add group IDs into the `supplementalGroups` to allow proper access.

### Motivation
Making in parallel with https://github.com/DataDog/integrations-core/pull/5212

hush-hush committed:
Remove test binary

Secret binary was commited by mistake

DylanLovesCoffee committed:
note RunAsAny in SCC

DylanLovesCoffee committed:
note RunAsAny in SCC

L3n41c committed:
Allow dots in cluster names (#4611)

* Allow dots in cluster names

because some users already have dots in their cluster names:
https://github.com/helm/charts/pull/19327#issuecomment-565535449

ogaca-dd committed:
Add 7.16.0-rc.9 and 6.16.0-rc.9 (#4613)

olivielpeau committed:
[release note] Add prelude note about Agent 7 (#4620)

olivielpeau committed:
Push "datadog/dogstatsd:7" tag as well when releasing latest (#4623)

ogaca-dd committed:
Add 6.16.0 and 7.16.0 (#4622)

* Add 6.16.0 and 7.16.0

* Use tags for INTEGRATIONS_CORE_VERSION and OMNIBUS_SOFTWARE_VERSION

hush-hush committed:
Adding golangci-lint to the tests

hush-hush committed:
Fixing most of the linting issue raised by golangci-lint

The old linter where run without build flags. This meant that any file
behing a build flag was not linted. This is why we have so many issue
when switching to golangci-lint.

ogaca-dd committed:
Update change log (#4629)

* Update CHANGELOG for 7.16.0

albertvaka committed:
install_script: warn if version not specified (#4438)

* install_script: warn if version not specified

Since we will install Agent 6 by default.

* Print warning in yellow

Co-Authored-By: Olivier Vielpeau <olivielpeau@users.noreply.github.com>

olivielpeau committed:
[macOS install script] Allow selecting major version to install (#4498)

* [mac_os] Allow selecting the major version of the DMG

Still default to 6

* [mac_os] Make install script fail if DMG can't be downloaded

albertvaka committed:
[BasicCache] Only update timestamp if value was added/changed (#4228)

And return a boolean to indicate what happened. Also changed the Get
function to mimic the behaviour of go maps.

Updated test.

remeh committed:
Properly install latest available versions of the Agent with install script on Fedora  (#4631)

* install_script: take care of `dnf install` default behavior which doesn't upgrade.

* install_script: dnf is available since fedora 22 which has been released in May 26, 201

* Update cmd/agent/install_script.sh

Co-Authored-By: Olivier Vielpeau <olivielpeau@users.noreply.github.com>

remeh committed:
docs: links for Agent v7 documentation. (#4610)

* docs: links for Agent v7 documentation.

* Apply suggestions from code review

Co-Authored-By: Pierre Guceski <pierre.guceski@datadoghq.com>

* Apply suggestions from code review

Co-Authored-By: Olivier Vielpeau <olivielpeau@users.noreply.github.com>

* Mention the Agent v7 in the README

KSerrania committed:
[kitchen] Use chef-datadog 4.0.0 (#4633)

arbll committed:
dsd parsing: remove allocations during int and float parsing (#4548)

unfortunately strconv.Parse* only accepts strings as parameters
this issue was reported at https://github.com/golang/go/issues/2632
but it does not seem that the Go team has a concrete plan to deal with it

we have two options to remove those allocs:
1. clone their parsing code or import a dependency that does that for us
2. use a dirty trick commonly used to temporarly convert []byte to string without allocs

arbll committed:
dsd aggregator: faster context key hashing  (#4461)

Commit:
Adding batching logic to retrieve External Metrics values (#4617)

* Remove logic to limit number of Autoscalers

* Add timeout to the Datadog Client

* Add logic to parallelise calls to the metrics backend

* Add tests

celenechang committed:
fix changelog typos (#4551)

celenechang committed:
[entrypoint] ensure leader election only if DD_LEADER_ELECTION is true (#4562)

celenechang committed:
[clcrunner] add pod ip to request header (#4587)

vboulineau committed:
Add container_mode to set source/service for container logs read from journald. (#4550)

It takes the value from tags and use a cache as it's called for every log line fetched from journald.
We randomize the TTL in cache to avoid stalling the agent for some time in case of huge number of containers/tags.

leeavital committed:
Automatically format tracer-ebpf.go (#4625)

We've been checking in an incorrectly formatted version of
tracer-ebpf.go for a while. This PR will automatically format it
when re-generated.

KSerrania committed:
[gitlab] Add details in failure slack message (#4555)

Add commit message and link to commit.

Commit:
Add cluster-name tag in host tags (#4558)

* Add cluster-name tage in host tags.

Attempt to extract the node label (`alpha.eksctl.io/cluster-name`) and use it for cluster-name.
Also use the `clustername.GetClusterName` (#3098) to extract via the cloud provider metadata.

* Refactor cluster_name change and remove dependency on clustername.GetClusterName from kubelet package.

As part of this commit moved the HostnameProvider to hostinfo package to avoid import cycle.

* Move node labels retrieval under kubeapiserver build tag

* Refactor further for better package layout

* Change GetKubeUtil to return an interface so that we can mock it in tests

* Correct GetNodeLabels => NodeLabels

* Further split use of apiserver for conditional build

* Fix signature to accept nodeName

* Fix ineff assignment

* Expose QueryKubelet - used in integration tests

* Update reference to apiserver.HostnameProvider in integration tests

* Fix missing imports for clustername, log

* Need both kubernetes/apiserver and hostname/apiserver

* Add release note

* Fix puppy build by aliasing kubelet to kubelet

* Remove the issues section from the release note

olivielpeau committed:
[gitlab] Tweak verbosity of jobs (#4642)

* remove listing of package file at the end of deb agent build,
  extremely verbose and of little value
* add `-e` parameter to all `inv deps` invocations, makes it much
  easier to understand where the command fails

olivielpeau committed:
[invoke] Add the v6 tag (if present) to the release title in changelog (#4637)

xlucas committed:
Explicitly pin to agent 7 (#4645)

mbakht committed:
[system-probe] Print kernel and platform versions when there are missing kernel functions (#4590)

* Print kernal and platform versions when there are missing kernel functions

* Always print sysprobe version, remove duplicate code, make error message more readable

* Modify log message to indicate the system probe has successfully started

kevinconaway committed:
Use DNS encoder in CollectorConnections (#4636)

* Use DNS encoder in CollectorConnections

* Fi mmh3 merge conflict

* Use DNS encoder in CollectorConnections

vboulineau committed:
Bump Debian base image to current testing (bullseye) (#4621)

NoisomePossum committed:
Allow ref to capture groups in mask replacement (#4563)

* Allow users to reference regex capture groups in mask sequence replacement

* Add release note

* Add test for when regex fails to match

ogaca-dd committed:
Document hostname_force_config_as_canonical (#4643)

L3n41c committed:
`inv test` learnt `--no-cache` to bypass go tests caching (#4544)

`go test` is caching the results of tests.
There can be some cases where we really want to run the same tests several times.
For example to investigate races or flaky tests.
In those cases, we can now call `inv test --no-cache` to bypass the test cache.

xlucas committed:
Dockerhub readmes (#4641)

leeavital committed:
Fix --incremental-build option for process-agent/system-probe (#4653)

We were using the `-i` flag to `go build` when passing
--incremental-build to the build system. `-i` causes the dependencies of
the build to be installed.

With inremental-build (system probe)
real	0m13.155s
user	0m3.032s
sys	0m6.796s

Without incremental build (system probe)
real	0m50.761s
user	0m32.740s
sys	0m12.024s

L3n41c committed:
Fix race condition in singleton initialization (#4545)

orisano committed:
docs: fix link syntax (#4655)

olivielpeau committed:
[docs] Update link to 7.16 hostname config option docs (#4659)

kbogtob committed:
Use open source ARM images for building on CI (#4654)

* Build the puppy agent on arm using OSS images

* Make build of ARM images more explicit

* Use ARM64 build images on all ARM64 builds

p-lambert committed:
[processes] Fix container check error handling (#4607)

p-lambert committed:
Always fallback to Host IP Provider (#4626)

oiooj committed:
[system-probe] Fix DNS cache name lost (#4597)

kbogtob committed:
Add dogstatsd ARM build (#4634)

* Regroup dogstatsd build tasks in .gitlab-ci.yml

* Rename puppy agent ARM build task

* Add dogstatsd ARM static & linked builds

* Add dogstatsd ARM binary check

Commit:
Add a tag for kubernetes node role in host metadata (#4594)

* Add a tag for kubernetes node role in host metadata

* Add release note

* Add tests for label to tags

* Address review feedback

clamoriniere commented on Dockerfiles/manifests/openshift/scc.yaml:
```diff
@@ -43,6 +43,8 @@ fsGroup:
 runAsUser:
+# With some versions of Openshift, you may need to run the agent as a root user to access the Docker socket. 
+# In that case, use `type: RunAsAny` instead
   type: MustRunAsRange
```
Another solution it to create a `group` that have access to the docker socket. then add this group in the`supplementalGroups`.
doc: https://docs.openshift.com/enterprise/3.1/install_config/persistent_storage/pod_security_context.html#supplemental-groups

DylanLovesCoffee replied:
Yes, this seems to be a better solution.

julien-lebot committed:
Added some documentation for building with onmnibus on Windows (#4556)

Added some documentation for building with onmnibus on Windows

julien-lebot committed:
Add documentation for Windows Dev Env (#4547)

Add documentation for Windows Dev Env

olivielpeau committed:
[invoke/rtloader] Add rtloader.clean command to clear CMake's cache (#4557)

Necessary when the path to some libraries detected by CMake have
changed.

Ideally, the command should clean up all build artifacts from rtloader,
this is out of the scope of this commit.

KSerrania committed:
[gitlab] Update Windows builders and fix env vars (#4674)

derekwbrown committed:
On Windows, when upgrading, installer will fail if the user attempts (#4670)

to assign a configuration file directory or binary directory that is
        different from the original.

derekwbrown committed:
Re-add windows source tests. (#4624)

* Re-add windows source tests.
Tests were removed because they're slow; however, with DAG feature
they won't block other stages

* Fix typos in .gitignore
Add protections from test-generated directories in .gitignore
Fix apm test (different line endings)

* Fix test parameters for x86

KSerrania committed:
[slack notifier] Add commit author (#4649)

KSerrania committed:
[kitchen] Revert allow failure on Windows upgrade from 6 jobs (#4671)

These tests should be working again, the latest Agent 6 version is 6.16.0 now.

leeavital committed:
Add an integration test for conntrack (#4483)

I had added scripts to setup and teardown iptables based NATing, but never . added the test that actually used the scripts. This PR adds a test which creates NAT'd UDP and TCP connections and makes sure the conntracker captures them.

L3n41c committed:
Fix tracer-ebpf compilation on Ubuntu 19.04 Disco (#4579)

Attempt to compile `system-probe` with Linux kernel headers 5.0 and LLVM 8
generates compilation error due to `asm goto`:
```
vagrant@ci-lenaic:~/go/src/github.com/DataDog/datadog-agent$ inv system-probe.build
In file included from ./pkg/ebpf/c/tracer-ebpf.c:17:
In file included from /usr/src/linux-headers-5.0.0-36-generic/include/linux/ptrace.h:6:
In file included from /usr/src/linux-headers-5.0.0-36-generic/include/linux/sched.h:12:
In file included from /usr/src/linux-headers-5.0.0-36-generic/arch/x86/include/asm/current.h:6:
In file included from /usr/src/linux-headers-5.0.0-36-generic/arch/x86/include/asm/percpu.h:45:
In file included from /usr/src/linux-headers-5.0.0-36-generic/include/linux/kernel.h:11:
In file included from /usr/src/linux-headers-5.0.0-36-generic/include/linux/bitops.h:19:
/usr/src/linux-headers-5.0.0-36-generic/arch/x86/include/asm/bitops.h:209:9: error: 'asm goto' constructs are not supported yet
        return GEN_BINARY_RMWcc(LOCK_PREFIX __ASM_SIZE(bts), *addr, c, "Ir", nr);
               ^
/usr/src/linux-headers-5.0.0-36-generic/arch/x86/include/asm/rmwcc.h:60:32: note: expanded from macro 'GEN_BINARY_RMWcc'
                               ^
/usr/src/linux-headers-5.0.0-36-generic/arch/x86/include/asm/rmwcc.h:10:28: note: expanded from macro 'RMWcc_CONCAT'
                           ^
/usr/src/linux-headers-5.0.0-36-generic/arch/x86/include/asm/rmwcc.h:9:30: note: expanded from macro '__RMWcc_CONCAT'
                             ^
note: (skipping 2 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/usr/src/linux-headers-5.0.0-36-generic/arch/x86/include/asm/rmwcc.h:54:2: note: expanded from macro 'GEN_BINARY_RMWcc_6'
        __GEN_RMWcc(op " %[val], " arg0, var, cc,                       \
        ^
/usr/src/linux-headers-5.0.0-36-generic/arch/x86/include/asm/rmwcc.h:21:2: note: expanded from macro '__GEN_RMWcc'
        asm_volatile_goto (fullop "; j" #cc " %l[cc_label]"             \
        ^
/usr/src/linux-headers-5.0.0-36-generic/include/linux/compiler_types.h:188:37: note: expanded from macro 'asm_volatile_goto'
                                    ^
```

`asm goto` have been required for x86 since Linux 4.17 but their use is spreading over versions.
So, with old enough Linux kernel headers, we wouldn’t have the above mentioned errors.

And support for `asm goto` has been added in LLVM 9:
https://prereleases.llvm.org/9.0.0/rc3/tools/clang/docs/ReleaseNotes.html#c-language-changes-in-clang
So, the above mentioned errors don’t happen with LLVM 9 or above.

A discussion about these compilation errors can be found there:
https://github.com/iovisor/bcc/issues/2119

As a workaround, if we have the “bad” combination of Linux kernel headers and LLVM,
we can apply the same trick as the one described in https://github.com/iovisor/bcc/issues/2119
in order to make the `asm goto` block compile if not used but make the compilation abort if used.

ahmed-mez committed:
[Logs][Fix] Simplify logs tagging logic + add waiting for k8s tags (#4661)

L3n41c committed:
Make the log agent retry to connect to docker rather than giving up (#4536)

This is to handle the case where the agent is started while the docker daemon is stopped
and the docker daemon is started later while the datadog agent is already running.

arbll committed:
dsd: rewrite parsing logic and split parsing vs enrichment (#4575)

mfpierre committed:
Bump required go version (#4678)

mfpierre committed:
Switch copyright to 2020 (#4679)

celenechang committed:
[dca] handle NewHandler failure better in setupClusterCheck (#4447)

* increase timeout to detect apiserver leader

* do not require LE to be running when setting up cluster checks

celenechang committed:
[docker] allow user to add custom docker label for tagging (#4648)

* [docker] allow user to add custom docker label for tagging

* add debug lines

leeavital committed:
Don't check that laddr.Ip is in local net.Interfaces when determining direction (#4628)

Commit:
Introduce Rate Limiting Stats in the /metrics of the Cluster Agent (#4669)

* Update datadog-go client library to get access to the rate limiting statistics

* add test suite

* refactor validateExternalMetric method

* add helper for the prometheus metrics

Commit:
Fix API endpoint (#4681)

* fix api prefix for rate limiting

ofek commented:
Any update?

mfpierre committed:
Harmonize copyright for 2020 (#4680)

DylanLovesCoffee committed:
include supplementalGroups

DylanLovesCoffee committed:
include supplementalGroups

arbll committed:
dsd udp: merge multiple datagrams into one "Packet" (#4573)

truthbk committed:
[changelog] updating changelog for 6.16.1/7.16.1.

includes omnibus-software note.

KSerrania committed:
[kitchen] Revert use of a specific Agent 5 version in 5->6/7 tests (#4672)

Reverts #4258, now that Agent 5 should uninstall correctly thanks to the trace-agent bugfix included in 5.32.5.

leeavital committed:
Fix eBPF codegen, make bytecode tests mandatory (#4682)

A previous change (#4683) caused empty object files to be created by the ebpf build step, and broke compilation on some dev VMs.

A more recent change tried to define asm_volatile_goto if it was not already
defined. The problem with this is subsequent header files (probably
compiler.h).

Bionic would not later try to redefine it, but older version of the
kernel would (e.g. my dev VM is running 4.4.0).

To get around this, include <linux/compiler.h> which should do the right
thing and make sure asm_volatile_goto is defined if necessary.

This change also makes the the bytecode test (which ensures the checked in object file matches the checked in C code) mandatory.

truthbk committed:
Merge branch 'master' of github.com:DataDog/datadog-agent

KSerrania committed:
Make agent version functions take an explicit major version parameter  (#4667)

Make all invoke functions depending on the major version take the major version parameter explicitly, instead of relying on environment variables.

Omnibus builds are somewhat trickier, as we can't pass parameters to bundle exec omnibus, only environment variables, and the omnibus builds do need the major version, as they call invoke functions. The way we're doing is thus as follows:
- inv agent.omnibus-build takes an explicit --major-version parameter,
- It injects this parameter to the omnibus env vars as `MAJOR_VERSION,
the omnibus software definitions use the MAJOR_VERSION env var to call the other invoke functions (such as inv agent.build, inv dogstatsd.build, etc.) with the explicit --major-version parameter.

This PR creates a tasks/winbuildscripts folder containing the windows build scripts, as this will become the preferred way to package the build scripts once #4614 is merged (instead of baking the scripts in the build image directly, which is inconvenient).

jahanzebk committed:
pkg/trace/stats: do not ignore single span sublayer metrics (#4571)

No longer ignoring childless spans for apm sublayer metrics and stats to fix inaccuracies in how sublayer metrics are sometimes reported.

It looks like this was done as an optimization since it makes sense if the service is always sending a single span, then the sublayer metrics are actually irrelevant. But in cases where the service sometimes has child spans and sometimes doesn't, this can cause the metric to simply be inaccurate since it doesn't take all metrics into account.

AlexandreYang committed:
Dogstatsd/Statsd Mapper Feature (#4471)

ogaca-dd committed:
Improve payloads events (#4184)

* Implement MarshalJSON for Events

remeh committed:
CODEOWNERS: add @agent-core to the codeowners of the Logs Agent. (#4688)

Commit:
Adding extra logic for container restart (#4686)

* adding extra logic for container restart

KSerrania committed:
[kitchen] Split tests on multiple Azure regions (#4689)

KSerrania committed:
[kitchen] Change SLES 15 image (#4690)

albertvaka committed:
Only import new RPM signing key when installing A7 (#4520)

We are only going to sign Agent 7 with the new key.

L3n41c committed:
Make circleci look for release notes in releasenotes-dca/notes (#4474)

so that cluster-agent specific PR don't need to add the `no-changelog` tag.

mfpierre committed:
Collect pods (#4406)

* Add GetRawLocalPodList

* Bootstrap new pod check

* Add scrubbing

* Add pod payload message

* Add chunking logic

* Add a new orchestrator tag only for the process agent

* Reuse existing cluster_name config setting

* Add logic for orchestrator endpoints

Commit:
Add read-secret command in cluster-agent to use as secrets backend (#4639)

* Add read-secret command in cluster-agent to use as secrets backend

* Address review feedback, fix formatting

* Check for os.IsNotExist

* Simplify errors to avoid issues across platforms

* Add readsecret script helper

* Correct symlink check logic

* Add release note

* Move read-secret command to a different package per review feedback

* Address review feedback, add tests for symlinks

* Undo edit in pkg/secrets/fetch_secret.go

* Address review feedback from @hush-hush

* Remove unused const

* Fix formatting in cmd/secrets/read_secret.go

* Address feedback from @hush-hush

* Adjust max secret sizes to make them more useful

- Max secret size = 8k
- Max JSON secret output = 1MB

* Change the max buffer size in config as well

* Duplicate var in no_secrets

* Rename `read-secret` command to `secret-helper read`

* Update command in the readsecret.sh

* Update copyright year to 2020 on new files

* Update config template with mention of new default

DylanLovesCoffee committed:
provide user example

DylanLovesCoffee committed:
provide user example

ahmed-mez committed:
show distribution metrics in agent commands (#4692)

kbogtob committed:
Fix SystemV debian init scripts for docker compatibility (#4691)

mfpierre committed:
Rely only on config to enable the pod check (#4695)

arbll committed:
dsd: pool metrics.MetricSample (#4630)+(#4696)

vboulineau committed:
Fix regression when matching containerID to Pod due to runtime prefixes (#4694)

KSerrania committed:
[kitchen] Change SLES 15 image again (#4697)

remeh committed:
telemetry: adds the telemetry package and related features. (#4599)

* telemetry: adds the telemetry package and related features.
* teletry: comment on the custom separator.
* telemetry: remove unused sync.Once in our telemetry Gauge/Counter.
* telemetry: add unit test for telemetry.IsCheckEnabled
* CODEOWNERS: the telemetry pkg belongs to the agent-core team
* telemetry: add a comment on billable custom metrics of the telemetry feature.
* telemetry: use a custom Prometheus registry.
* telemetry: comment public method.
* telemetry: better naming of tags parameters.
* telemetry: add observability metrics using the telemetry feature. (#4600)
* telemetry: adds the telemetry package and related features.
* telemetry: add observability metrics using the telemetry feature.
* telemetry: linter compliant.
* telemetry: metrics samples is a counter.
* telemetry: use constants for Ok / Error states.
* telemetry: optional subsystem + address missing metrics.
* telemetry: use new signature of NewProcessCollector since we've updated the prom dep.
* telemetry: fixing autoscalers unit tests.
* Gopkg.lock: remove unused promutil lib from dependencies.

p-lambert committed:
[netlink] Add double entries to conntracker cache to support SNAT (#4656)

* [netlink] Add double entries to conntracker cache to support SNAT connections

* [netlink] Measure insertion time for both tuples

L3n41c committed:
Add hostname sanity checks to prevent using POD name (#4503)

Try to detect if the agent is running in a container with a UTS namespace which is not the root one.
In those cases, the output of `hostname -f` doesn’t reflect the identity of the host and shouldn’t be used.

gbbr committed:
pkg/trace/agent: use shared agent logger (#4693)

This change removes the custom logger implementation and switches to
using the shared library logger. As a result, support for
`apm_config.log_throttling` is removed and instead we gain support for
full logging features such as `log_to_console` and `log_to_syslog`.

Log throttling is not something that users need to manually configure
and should be taken care of by developers to ensure that errors and
warnings never spam the user. A new package `pkg/trace/logutil` has
been added which exposes a throttled logger that allows to limit how
many times a certain message appears during a time window.

gbbr committed:
pkg/trace/obfuscate: add a filter to extract table names from queries (#4399)

### Description

This change introduces a new tag called `sql.tables` which contains the tables that a query addresses as list of comma-separated values. To enable it, the feature flag `table_names` needs to be set using `DD_APM_FEATURES=table_names`. For example, given the query:

```
SELECT * FROM users JOIN friends
```

Will result in the tag `sql.tables` to contain the values `users,friends`

### Limitations

The current algorithm works agnostic of SQL engine. It simply takes the first word that follows any of the following keywords: `FROM`, `UPDATE`, `INTO` and `JOIN`. 

* Different SQL engines may have different syntax and could contain further keywords ([example](https://www.postgresql.org/docs/9.5/sql-select.html)) after these ones which are not table names. This case is not covered by the current implementation.

* If we later on decide to have the tracing clients mark the SQL engine as a tag, we could use it to further improve the processing.

L3n41c committed:
Fix master (#4701)

olivielpeau committed:
[codeowners] Change ownership of config_template file (#4570)

* Baklava since the config template is documentation
* Agent-all as any agent team has ownership to update this file

olivielpeau committed:
Revert "Revert "[rpm] Delete removed local files from remote staging repo (#4416)" (#4524)" (#4525)

This reverts commit 51491762e4372c3c42d70afb9efab113e949d600.

xlucas committed:
Update internal tagging (#4336)

knusbaum committed:
pkg/config: add datadog.tracer to the list of StandardStatsdPrefixes for tracer health metrics. (#4588)

We are adding health metrics to the tracers. (see: DataDog/dd-trace-go#534)

We need to whitelist stats starting with datadog.tracer

L3n41c commented on Dockerfiles/manifests/openshift/scc.yaml:
```diff
@@ -42,10 +42,18 @@ defaultAddCapabilities: []
   type: MustRunAs
 readOnlyRootFilesystem: false
-runAsUser:
+runAsUser: 
```
Extra trailing whitespace

L3n41c commented on Dockerfiles/manifests/openshift/scc.yaml:
```diff
@@ -42,10 +42,18 @@ defaultAddCapabilities: []
   type: RunAsAny
+# If your environment restricts user access to the Docker socket, 
+# you can add a group under `supplementalGroups` that has access
+# by uncommenting the following lines (and removing the previous line `type: RunAsAny`):
```
If I remember well how RHEL and OpenShift are working, by default, there is no `docker` group defined on the system and the docker socket belongs to `root:root`.
The end-user needs to create a `docker` group (with `groupadd`) and restart the docker daemon so that it recreates a socket that belongs to `root:docker`.
Once this has been done, the SCC can be enhanced to add the `SupplementalGroups`.

I don’t think it is desirable to have the `docker` group creation done by the Datadog deployment. But it could be smart to put in the comment the fact that the end-user has this setup to do on the host before hand.

DylanLovesCoffee replied:
Ah, good point. I'll generalize this section a bit more as this is dependent on the environment and should be managed by the user according to their setup.

arbll committed:
aggregator: improvements to the deduplication algorithm (#4644)

JM-Lazaro committed:
Fix to Allow Mounting Empty Directory to Disable Docker Check (#4349)

ogaca-dd committed:
Set enable_events_stream_payload_serialization at True (#4703)

ogaca-dd committed:
[Doc]: Explain how to run the agent. (#4603)

[Doc]: Explain how to run the agent.

hush-hush committed:
Update builders to allow building updated software deps (#4657)

Note: the windows builder images do not need to be updated.
Their updated image builds have silently failed, so keeping them unchanged for now.

Co-authored-by: Olivier Vielpeau <olivielpeau@users.noreply.github.com>

derekwbrown committed:
Create new artifacts for cloudfoundry support (#4614)

* WIP - Do Not Merge

Adds new zip file target in the binary stage with puppy & apm for
windows (for now).

Adds new zip file in the package stage, which is the zip of the entire
windows package.

* Merge 7.16/master

* Add dogstatsd to buildpack build
Zip multiple directories

* WIP - Do Not Merge

Adds new zip file target in the binary stage with puppy & apm for
windows (for now).

Adds new zip file in the package stage, which is the zip of the entire
windows package.

* Add dogstatsd to buildpack build
Zip multiple directories

* Refactor the custom action library a bit so that code could be shared
with the command line installer.

Initial rev of command line executable that does all the steps that the
CAL does (for use with the BOSH release)

* Add command line parsing to allow install script to call install-cmd
with correct options

* Create a new MSI that's just dogstatsd.
(doesn't quite work yet, can't find config file)

* Fix dogstatsd to actually work on windows.
Find windows config location
Run as service (local_service).  Set proper file/registry permissions

* Adds puppy build
- puppy builds 32 bit msi
- puppy builds 64 bit msi
- converge .gitlab-ci.yml down for less cut/paste

* Review feedback

* Actually run the custom action from install-cmd

* More review feedback I missed earlier

* Fix zip file layout to be more inline with the installed binaries

* Clean up buildpack zip file creation

* Fix rebase error(s)

* Latest review feedback

* More review feedback

* Fix vetting error

* Set omnibus-ruby back to `datadog-5.5.0`

xlucas committed:
Collect resource tags on ECS (#4387)

p-lambert committed:
[system-probe] Add support for docker-proxy traffic (#4665)

* [system-probe] Filter docker-proxy connections

* Resolve the raddr side of connections for local container

p-lambert committed:
[process-agent] Fix behavior when custom endpoint has a path specified (#4540)

p-lambert committed:
[system-probe] Allow system-probe `systemd` to be enabled (auto-initiated) (#4591)

p-lambert committed:
[system-probe] Enable DNS inspection by default (#4705)

* [system-probe] Enable DNS inspection by default

* Add a releasenote

Co-authored-by: Lee Avital <leeavital@gmail.com>

leeavital committed:
[system-probe] Fix conntrack test (#4704)

The old test was using the same local address for both a NATed connection, and
a non-NATed connection.

This changes the test to create separate connections that are NATed and
non-NATed

Commit:
Logic to fully support AWS Fargate on Amazon EKS (#4699)

* Adding logic to only schedule the kubelet and the eks_fargate check

* Adding logic to generate kubelet endpoint through the APIServer 

Co-authored-by: Ahmed Mezghani <38987709+ahmed-mez@users.noreply.github.com>

derekwbrown committed:
Fix registration of agent as event log source (#4706)

truthbk committed:
[jmxfetch] bump to 0.34.0 (#4709)

* [release] bumping jmxfetch to 0.34.0

* [reno] adding release note.

julien-lebot committed:
Add 7.17.0-rc.1 and 6.17.0-rc.1 (#4710)

KSerrania committed:
Update Windows builders to use latest ffi-yajl (#4712)

KSerrania committed:
Revert "Revert "Revert "[rpm] Delete removed local files from remote staging repo (#4416)" (#4524)" (#4525)" (#4714)

This reverts commit 313cbc5c383cab4f7fa54002a82131c7749c74bf.

DylanLovesCoffee committed:
rewording

DylanLovesCoffee committed:
rewording

julien-lebot committed:
Use custom images for kitchen tests (#4708)

Use custom images in kitchen tests: Windows 2008 R2 & SUSE 11 SP4 are EOL.

Co-authored-by: Kylian Serrania <kylian.serrania@datadoghq.com>

shang-wang committed:
remove exposing config content from expvar for process-agent (#4713)

Commit:
Fix secret helper read secret test Windows  (#4711)

* Add .gitattributes config to normalize line endings to LF

* Trying with -text

* Hack around the problem by removing end of line in test fixture file

derekwbrown committed:
Fix build cancellation race condition by having transfer directory (#4717)

between the docker image and the host include the job ID in the path

Also, reduce copy/paste error by consolidating all of the build batch
files into one.

KSerrania committed:
[omnibus] Codesign binaries shipped in MacOS package (#4722)

kbogtob committed:
Add 7.17.0-rc.2 and 6.17.0-rc.2 (#4724)

leeavital committed:
[system-probe] use IsLoopback instead of Direction=LOCAL to filter local DNS (#4727)

github.com/DataDog/datadog-agent/pull/4628 got rid of the LOCAL
connection direction which had the unfortunate side effect of
breaking how we enforce the CollectLocalDNS flag.

We were using {s,d}port==53 && direction==LOCAL to filter DNS. Now we will use
{s,d}port==53 && dest.IsLoopback.

As part of this, add Address#IsLoopback uses net.IP#IsLoopback.

remeh committed:
telemetry: do not set the namespace, the configuration check will take care of it (#4730)

* telemetry: different namespace for the agent and the cluster-agent.
* telemetry: do not set a namespace, the check configuration will take care of it.

remeh committed:
telemetry: a metric of the cluster agent was not properly ported to the telemetry pkg. (#4733)

prognant committed:
Bump copyright year to 2020 (#4721)

julien-lebot committed:
Create 7.17 RC3 (#4734)

Commit:
Check if CheckMetadata exist before iterating over it in cluster agent status page (#4728)

clamoriniere committed:
[cluster-agent] fix eventrecorder init (#4732)

vboulineau committed:
Fix kube node roles to host tags feature (handles multiple roles / node) (#4731)

celenechang committed:
Revert "[docker] allow user to add custom docker label for tagging (#4648)" (#4737)

This reverts commit ba3112fd4bc021908cab5206ef8ca8e2475ac74d.

remeh committed:
telemetry: avoid using the `type` tag. (#4740)

* telemetry: avoid using the `type` tag.
* Better naming of the type tag.

Co-authored-by: Olivier Vielpeau <olivielpeau@users.noreply.github.com>

gbbr committed:
Revert "pkg/trace/stats: do not ignore single span sublayer metrics (#4571)" (#4751)

This reverts commit 1fe673f53805980b9854ab8595e105630277eaa4 from #4571.

The increase in traffic on our infrastructure needs to be tested more
before we can release confidently.

remeh committed:
telemetry: support options while creating telemetry metrics. (#4742)

* telemetry: support options while creating telemetry metrics.
* telemetry: lint.
* telemetry: typo.

Commit:
Adding logic to show DCA status for clc (#4738)

ogaca-dd committed:
Fix typo in writeEvent (#4725)

arbll committed:
dsd: add back missing packet return to the pool (#4748)

KSerrania committed:
[kitchen] Use forked driver with retry logic (#4747)

Uses the forked kitchen driver (https://github.com/DataDog/kitchen-azurerm) which adds a retry logic on VM creation & deletion.

arbll committed:
dsd: remove one alloc by sample linked to telemetry (#4755)

albertvaka committed:
Gitlab DAG: Add more dependencies (#4736)

- agent_rpm needs run_tests_rpm
- Split A6 and A7 kitchen tests
- agent_suse needs run_tests_rpm
- Dogstatsd package build needs on dogstats binary build
- Wait until we know the binary itself builds before running omnibus
- Move cloudfoundry buildpack from binary_build to package_build
- Deduplicate kitchen stuff
- Make agent-based builds need agent tests

remeh committed:
omnibus: bump python3 to 3.7.6 on Linux/macOS/Windows (#4741)

* omnibus: bump python3 to 3.7.6

* general: add release note entry for Python 3.7.6

* omnibus/python3: update .zip checksum.

* Fix x86 sha256 for new upload

Co-authored-by: Derek Brown <derek.brown@datadoghq.com>

olivielpeau committed:
[dogstatsd] Include full metric message in error when invalid (#4760)

So that the full metric message is logged when it's invalid. Normalize
how we log invalid events and service check messages as well.

albertvaka committed:
Make kitchen gitlab-ci task (#4757)

L3n41c committed:
CI: Add `--signed-push` to `docker.publish-bulk` for `datadog/agent-ARCH` (#4685)

KSerrania committed:
Revert "Gitlab DAG: Add more dependencies (#4736)" (#4761)

This reverts commit 3bc84fb621f1db1d51deb85ca30333ab23fd8496.

albertvaka committed:
Revert "Revert "Gitlab DAG: Add more dependencies (#4736)" (#4761)"

This reverts commit 9895bea30c3a1b1dc5f3a9cd067ae8dbc50d3a23.

albertvaka committed:
test

albertvaka committed:
Oops, revert two last commits pushed by mistake

julien-lebot committed:
Make 7.17 RC4 (#4762)

albertvaka committed:
Fix WINDOWS_TESTING_S3_BUCKET not being defined (#4764)

* Revert "Revert "Gitlab DAG: Add more dependencies (#4736)" (#4761)"

This reverts commit 9895bea30c3a1b1dc5f3a9cd067ae8dbc50d3a23.

* Fix WINDOWS_TESTING_S3_BUCKET not being defined

KSerrania committed:
Fix inv make-kitchen-gitlab-yaml to correctly render python runtimes var (#4765)

prognant committed:
Adds additional log_level documentation in config_template.yaml (#4723)

Two additions, one listing all the valid log_level and a note
about mandatory quotes for the `off` log level.

prognant committed:
Add checks last execution date in status output (#4743)

leeavital committed:
Fix flaky test in conntracker_test.go (#4746)

We recently made a change in cconntracker.go that puts two connections
(instead of one) in realConntracker.state for each register() call.

One of the tests wrote code assuming that there was only one entry,
like:

for _, r := range ct.state {
  break
}

Because for loops in go provide random iteration order, the test would
sometimes fail.

derekwbrown committed:
Fix build script for puppy, DSD, and cf_buildpack build. (#4768)

Wasn't parsing the environment variables right

albertvaka committed:
[gitlab] Do not trigger pipeline on tag creation (#4758)

* Cancel tag pipelines if not triggered via webhook

* Workaround gitlab issue

Where jobs would run even if their dependencies are skipped.

hush-hush committed:
Splitting golangci task to remain under circleCI memory limit

remeh committed:
telemetry: fix an error between UDS/UDP packets bytes count. (#4769)

* telemetry: fix error in dogstatsd telemetry;

* telemetry: remove two unused vars.

remeh committed:
telemetry: proper tags for this telemetry metric. (#4773)

KSerrania committed:
Revert "[gitlab] Do not trigger pipeline on tag creation (#4758)" (#4776)

This reverts commit 7c25c2ed8484397e2c8b787ceb88d8c8c9a110a7.
The workaround we used (adding a needs on the fail_on_non_triggered_tag job) makes the whole pipeline disappear on a regular branch pipeline (and makes web pipelines crash), because:
- fail_on_non_triggered_tag is skipped, thus the whole fail_on_tag stage is skipped. As a consequence, the stage is removed from the pipeline,
- then, the jobs which need fail_on_non_triggered_tag error out, since the needed job does not exist.

vboulineau committed:
Fix golangci issues for container stuff (#4735)

truthbk committed:
[invoke] add golangci-lint to the bootstrappable deps (#4783)

albertvaka committed:
Fix "Do not trigger pipeline on tag creation" (#4781)

* Revert "Revert "[gitlab] Do not trigger pipeline on tag creation (#4758)" (#4776)"

This reverts commit d925a9559f93b0a3abcc73348d7d58265d4d0860.

* Run always, chek for tags in the script

julien-lebot committed:
Add create_rc invoke task (#4767)

clamoriniere committed:
update go-datadog-api dep for DCA fix (#4720)

derekwbrown committed:
Update to use new gohai for > 64 cpu fix (#4766)

Add releasenote

Switch back to master

Commit:
Correct Python3 syntax in /readsecret.py (#4789)

* Fixes #4650 [v7] Syntax in /readsecret.py for Py3

- Correct syntax in /readsecret.py helper for DataDog Agent v7 (Python3).
- Fix test (different exception raised).
- Address minor deprecation warnings in tests.

* Add release note

* Fix the test task

Commit:
Check for missing WPA CRD before starting WPA controller loop (#4744)

* Wait for WPA CRD to become available before starting WPA controller loop

* Shorten filename

* Address review feedback

- Add the next retry delay time in log trace.

* Remove delay randomization, clean up log trace

* Fix the problem with Informer initialization for WPA

- Move start of the informer to the WPA check goroutine to fix the problem.
- Cleanup and code review feedback
- TODO for later to refactor the shared informer factory initialization.

* Remove a sub-package from Gopkg.lock

Commit:
Fix dead code flagged by golangci in wpa_controller.go (#4797)

p-lambert committed:
[ebpf] Add log entry when system-probe isn't running on root netns (#4790)

ramkrishankumar committed:
Remove explicit check of sketch memsize to 32Bytes since it depends on the underlying arch (#4792)

KSerrania committed:
[kitchen] Fix issues with SLES kitchen tests (#4799)

Updates SLES 12 Azure image.
Fixes SLES A7 kitchen scenario (was using the A6 config).
Fixes step-by-step command (copy-paste error from the install script, $repo_url is not defined).

felixb committed:
ignore missing label com.datadoghq.ad.check_names (#4775)

KSerrania committed:
Replace uses of the PYTHON_RUNTIMES env var by explicit parameter (#4702)

Makes invoke functions take the python runtimes list as an explicit parameter rather than an environment variable.
The PYTHON_RUNTIMES environment variable remains in the .gitlab-ci.yml file as it's used when multiple build jobs are using the same base build job, to pass the specific job parameters.

olivielpeau committed:
[dogstatsd] Continue parsing packets with consecutive newlines (#4805)

When more than 1 consecutive newline was found, the logic would ignore
the rest of the packet. This worked fine so far probably because there
is no dogstatsd datagram out there that is badly formatted enough to
have 2 consecutive newlines in their contents.

That said, since the addition of the buffer of multiple packets, if a
small datagram has a trailing newline, the next small datagram can end
up appended to it with an additional newline separating the 2. In that
case, the parser will then see into 2 consecutive newlines. If it stops
parsing the packet at that point, it will have ignored all the messages
in the second datagram.

Fix this by continuing to parse the packet even after 2 or more
consecutive newlines are found.

KSerrania committed:
[kitchen] Add A7->A7 upgrade tests (#4808)

Adds Agent 7 to Agent 7 upgrade scenario to kitchen tests.

Co-Authored-By: Albert Vaca <albert.vaca@datadoghq.com>

leeavital committed:
[ebpf] fix alignment of dns_cache telemetry fields (#4803)

Commit:
Add intrahost classification algorithm for all connections (#4698)

This change will have the agent begin classifying connection traffic as intrahost if the start and destination addresses are equivalent. It accounts for NAT and loopback interfaces, and adds a field to the agent payload so that the information is propagated to Datadog's backend. 

Co-authored-by: Lee Avital <leeavital@gmail.com>

sunhay committed:
Update CODEOWNERS for processes/networks split (#4804)

DylanLovesCoffee committed:
add journald mention

DylanLovesCoffee committed:
add journald mention

vboulineau committed:
Increase CircleCI VM Size for unit tests (#4784)

olivielpeau committed:
[reno] Change title of security-related section (#4716)

Switch from "security issues" to the more neutral "security notes"

arbll committed:
sort uniq: benchmark fix (#4719)

albertvaka committed:
Simplify kitchen test invokation (#4815)

* kitchen diagnose is already run by run-tests-kitchen

No need for a separate script.

* apt-test-6.sh doesn't seem to be used

* Remove unnecessary usage of rake

It was only used to execute shell commands

* kitchen-apt-check-azure.yml was only used in apt-test-6

albertvaka committed:
Fix .pyc files clenaup (#4814)

- `$INSTALL_DIR/agent` doesn't exist
- Instead, most `.pyc` files will be in `$INSTALL_DIR/embedded/lib/python*`
- We don't need to print an error if no `.pyc` files are found

remeh committed:
Upgrade the Agent build to embed Python 3.8.1 (#4464)

* Upgrade to Python 3.8
* Update Python 3.8 hash (Linux).
* Update python3 shasum for windows.
* Patch must be applied to Python 3.8 now.
* Update sha256 to reflect new zips
* omnibus: update pip3.
We've a bug while building python3 due to this:
  - https://github.com/pypa/wheel/pull/303
We need pip3 >= `19.2.3`.
* omnibus: in order to use a more recent pip, we need a more recent pip-tools.
pip >= 19.3 support landed in pip-tools 4.2.0
* Release note entry.
* Apply suggestions from code review
Co-Authored-By: ruthnaebeck <19349244+ruthnaebeck@users.noreply.github.com>
* Update releasenotes/notes/python-3-8-c66cde69f972eb1e.yaml
Co-Authored-By: ruthnaebeck <19349244+ruthnaebeck@users.noreply.github.com>
* tasks: use `distro` instead of `platform` to get running linux distribution.
* gitlab-ci: install requirements deps for invoke tasks.
* Revert "gitlab-ci: install requirements deps for invoke tasks."
This reverts commit 9a6b8c78734bcdf946c8318678a2c0f3d2a663f1.
* gitlab-ci: use new builders shipping `distro` dep for invoke tasks.
* gitlab-ci: ebpf tests also need the distro dep
* gitlab-ci: builders images.
* general: update python 3 to 3.8.1
* gitlab-ci: install `distro` dep for python 3 in tests ebpf job.
* gitlab-ci: merge WINBUILDIMAGES and BUILDERS vars.
* omnibus: python3.8.1 checksum.
* gitlab-ci: merge ARMBUILDIMAGES and BUILDIMAGES env vars.
* omnibus/python3: update sha256sum of windows archives.
* empty commit, can't find my pipeline
* tasks: python 3.8 compliant.
See https://docs.python.org/3/whatsnew/3.8.html#porting-to-python-3-8
* omnibus/python3: update wheel dep to 0.34.1
* appveyor: configure appveyor to use python 3.8
* python3: updated python 3.8.1 32bits for windows sha256sum
* omnibus/pkg-scripts: remove downloader data for both 3.7 & 3.8 pythons
* omnibus: remove pythons download data at the right moment only.
* omnibus: properly call remove_downlaoder_data for centos.
* gitlab: use new builders images.
* gitlab: use builders images merged in master.
* gitlab-ci: use the requirements.txt while installing distro for the ebpf part
* tasks/build_tags: proper entry for redhat since we switched to distro.id()
* packages-scripts: also remove python 3.8 in preinst
* Revert "omnibus: properly call remove_downlaoder_data for centos."
This reverts commit 427eba39d03dfb52134b79c6454882f4faf72feb.
* releasesnotes: update python 3.8.1 release note entry.
* Update releasenotes/notes/python-3-8-c66cde69f972eb1e.yaml
Co-Authored-By: Olivier Vielpeau <olivielpeau@users.noreply.github.com>
* releasenotes: adapt python 3.8.1 release note entry.
* releasenotes: remove mention of a7 py version.
Co-authored-by: Derek Brown <derek.brown@datadoghq.com>
Co-authored-by: ruthnaebeck <19349244+ruthnaebeck@users.noreply.github.com>
Co-authored-by: Olivier Vielpeau <olivielpeau@users.noreply.github.com>

julien-lebot committed:
Build final artifacts for 7.17.0 (#4820)

julien-lebot committed:
Fix jmxfetch hash (#4824)

olivielpeau committed:
[logs/registry] Do not log error if registry does not exist (#4763)

It's expected for the registry to not exist when the agent starts for
the first time, so only log at the debug level when that happens.

remeh committed:
cmd/jmx: properly pass the loglevel to the JMX runner. (#4828)

* cmd/jmx: properly pass the loglevel to the JMX runner.
* cmd/jmx: add release note entry

julien-lebot committed:
Backport changelog to master for release 7.17.0 (#4835)

Co-authored-by: Kylian Serrania <kylian.serrania@datadoghq.com>

vboulineau committed:
Add container uptime metric for Docker/Containerd/CRI (#4753)

olivielpeau committed:
[ec2/test] Increase timeout on unit tests of ec2 util (#4817)

To try to avoid flakes on appveyor

prognant committed:
Correctly relocate min_collection_interval settings during A5->A6/7 config conversion (#4809)

clamoriniere committed:
[DCA] fix cluster-agent flare panic (#4838)

if the cluster-agent api was not reachable, the flare command returns
a Panic error message.

Solution: in case of cluster-agent connection error when retrieving the
`MetadataMapBundle`
* the function `convertmetadataMapperBundleToAPI` is not executed anymore
* in `convertmetadataMapperBundleToAPI()` function add a check to validate
  that the input is not `nil`

vboulineau committed:
Remove setcap NET_BIND_SERVICE as we cannot make it work with user namespaces used in the CI (#4846)

dmarkey committed:
Fix type (#4831)

vboulineau committed:
Add service listener in endpoints to watch for newly annotated services (#4816)

DylanLovesCoffee committed:
rebase needed
